### ISSUE TITLE
Layered algorithm in cytoscape overlaps edges and nodes

### ISSUE THREAD

**Issue by icd2k3** [cytoscape/cytoscape.js-elk#47]
Hi there!

Looking at the image for "Layered" [in the ELK docs](https://eclipse.dev/elk/reference/algorithms/org-eclipse-elk-layered.html), it seems that edges should not overlap each other, and also avoid overlapping nodes:
<img width="433" alt="image" src="https://github.com/cytoscape/cytoscape.js-elk/assets/1458959/9ffbba0b-5be0-43b9-a4a7-b65ec68c1035">

However, it seems it is not possible to achieve the same behavior in cytoscape? There is a similar question in Stack Overflow [here](https://stackoverflow.com/questions/66786313/keep-edges-seperated) that seems to suggest it is not possible either?

**Comment by maxkfranz** [comment]
> However, it seems it is not possible to achieve the same behavior in cytoscape?

Everything is possible in Cytoscape.  Just style it and use it the way you want.

The [documentation](https://js.cytoscape.org) and [Phind](https://www.phind.com/search?c=I%27m%20using%20the%20Cytoscape.js%20graph%20theory%20JS%20library.&q=How%20do%20I%20create%20a%20graph%20in%20my%20HTML%20page) have everything you need.

**Comment by fabric-io-rodrigues** [comment]
I tried some ways to make the curves generated by ELK in cytoscapejs. I had to use the ELK algorithm directly to get the list of points of the lines. I used segment-distances and segment-weights to define the intermediate points.

![Image](https://github.com/user-attachments/assets/723edde0-4bda-485b-8bcc-66e43268a731)

I didn't find it very easy, and it didn't turn out 100% good. If there was a way to draw the lines on the canvas and still maintain the same mouse hover/click functionality and style definition, that would be great.

If any adventurer wants to see the code that I managed to close and maybe generate a new version, adjusting whatever is necessary, feel free.

Here's the link: (https://gist.github.com/fabric-io-rodrigues/f5304881e354bfc99f3623519567e4cc)
